### ISSUE TITLE
Web API - Possible Breaking Change After Upgrade in App.Data.Update

### ISSUE THREAD

**Issue by tpperlman** [2sic/2sxc#2696]
**I'm submitting a ...**
[x] bug report
[x] not sure

**...about**
[x] Content Types or data management
[x] APIs like REST

**Current behavior**
Our production site runs on DNN v09.08.01 with 2SXC v13.1.0. We have written a Web API for customers to create their own DNN account and then updates 2SXC Customer data with the newly created UserID from DNN. The Web API accepts unauthenticated requests and has been functioning without issue.

The Web API currently inherits from Custom.Dnn.Api12. It creates a reference to a separate CS file that inherits from ToSic.Sxc.Dnn.DynamicCode. After the user account is created in DNN, we perform the following 2SXC update:

```
var updateOwner = new Dictionary<string, object>();
updateOwner.Add("EmailAddress", emailAddress);
updateOwner.Add("UserID", objUser.UserID);
App.Data.Update(owner.EntityId, updateOwner);
```

We upgraded our development environment to DNN v09.10.02 and 2SXC v13.4.0. The Web API still inherits from Custom.Dnn.Api12, but now the referenced CS file inherits from Custom.Dnn.Code12. When we run the same APIs we get the following error in the log:

> User is not allowed to update published or draft entity.

We're not sure if the previously installed 2SXC version was incorrectly allowing us to update data via the Web API in the referenced method, or if a security setting changed that does not allow us to update existing data. We are aware that 2SXC runs in a user context, and are hesitant to make a permissions change to the data that will open up all records to be modified via the REST API by unauthenticated users. We "could" also do a direct update of the entities in the database via SQL, but that isn't ideal for a number of reasons. Ideally we'd like to stick within 2SXC to resolve this.

* **2sxc version(s):** 13.4.0
* **Browser:** all
* **DNN:** 09.10.02
* **Language:** all

**Comment by iJungleboy** [comment]
If you update data through your C# code the permissions don't apply. They should only apply to REST apis. 

We did make some permission changes because we noticed something in REST wasn't quite right, maybe we accidentally added this too deep. 

@tvatavuk can you please check and then discuss?

**Comment by tpperlman** [comment]
> If you update data through your C# code the permissions don't apply. They should only apply to REST apis.

@iJungleboy thank you for confirming that! I'd been scouring docs.2sxc.org and hadn't been able to confirm that.

We appreciate the acknowledgement and quick response on this!

**Comment by iJungleboy** [comment]
@tpperlman should be fixed in 13.05 ;)
